code/game/objects/structures/spawner.dm

```diff
@@ -147,3 +147,8 @@ GLOBAL_LIST_INIT(astroloot, list(
 	name = "wumborian fugu den"
 	desc = "A den housing a nest of wumborian fugus, how do they all even fit in there?"
 	mob_types = list(/mob/living/simple_animal/hostile/asteroid/fugu)
+
+/obj/structure/spawner/mining/carp
+	name = "carp den"
+	desc = "A den housing a nest of space carp, seems fishy!"
+	mob_types = list(/mob/living/simple_animal/hostile/carp)
```

Comment:
please look at making some presets (like how other planets do their spawners)|
Also note you can add a custom spawn sound